### PR TITLE
User update endpoint

### PR DESCRIPTION
**Pull request by DANLENEHAN**
ToDo:

- Add user deletion endpoint
- Properly handle validation errors.

### CODE REVIEW

File: grau/db/user/user_model.py [hunk 1]
```diff
@@ -58,65 +58,89 @@ def get_id(self) -> str:
 
         return encrypt_str(str(self.session_id))
 
+    def to_dict(self) -> dict:
```
Comment (DanMaguireData) [inline grau/db/user/user_model.py:61]: Why are these fields excluded from the to_dict method?

**Review by DanMaguireData — COMMENTED** [review]
Those are the only things i have, g2g after that

File: grau/db/user/user_model.py (hunk 1 above)
Comment (DANLENEHAN) [reply grau/db/user/user_model.py:61]: This are the fields we don't want to send back on the details endpoint. I guess I could remove the exclude remove here and apply that logic in a function outside of the user_model?

File: tests/test_grau/blueprints/user/test_functions.py [hunk 2]
```diff
@@ -131,3 +131,42 @@ def test_attempt_logout_no_user(self, db_session):
...
+        )
+
+        # When
+        result = functions.update_user(
+            db_session=db_session,
+            user_dict=user_dict,
+            user=user,
+        )
+        # Then
+        assert result["weight_unit_pref"] == user_dict["weight_unit_pref"]
```
[the remark this answers is not in the document]
Comment (DANLENEHAN) [reply tests/test_grau/blueprints/user/test_functions.py:157]: Yeah we could I'll update that.

**Pull request merged**